bank/src/main.rs

```diff
@@ -45,29 +53,63 @@ fn transfer_funds(txn: &mut Transaction, from: i64, to: i64, amount: i64) -> Res
     Ok(())
 }
 
-fn ssl_config() -> Result<MakeTlsConnector, ErrorStack> {
-    let mut builder = SslConnector::builder(SslMethod::tls())?;
-    builder.set_ca_file("certs/ca.crt")?;
-    builder.set_certificate_chain_file("certs/client.maxroach.crt")?;
-    builder.set_private_key_file("certs/client.maxroach.key", SslFiletype::PEM)?;
-    Ok(MakeTlsConnector::new(builder.build()))
+fn delete_accounts(txn: &mut Transaction) -> Result<(), Error> {
+    txn.execute(
+        "DELETE FROM accounts", &[],
+    )?;
+    Ok(())
 }
 
 fn main() {
     let connector = ssl_config().unwrap();
+    let connection_uri = env!("DATABASE_URL");
     let mut client =
-        Client::connect("postgresql://maxroach@localhost:26257/bank", connector).unwrap();
+        Client::connect(&connection_uri, connector).unwrap();
 
+    // Create the "accounts" table.
+    client
+        .execute(
+            "CREATE TABLE IF NOT EXISTS accounts (id UUID PRIMARY KEY, balance INT)",
+            &[],
+        )
+        .unwrap();
+
+    let id1 = Uuid::new_v4();
```

Comment:
I matched the rest of the other simple CRUD apps and am generating it client-side, mostly because it avoids having to make another `SELECT` call to get the new IDs. But if you think it's better to use the server function, I can implement that.

Reply:
Yeah, I've seen this pattern in our other apps but I think we work towards getting rid of it. The reason that we encourage server-side is because CockroachDB can safely assume that a server-generated UUID is unique, meaning it doesn't need to read from the database when inserting a new row. However, if the UUID is generated on the client, then the server can't assume that it's unique so it must check for uniqueness. The cost of doing this, especially in a multi-region cluster, can be substantial.

It should be possible to avoid the extra query by using `RETURNING id` when inserting.

Reply:
I forgot about `RETURNING`. Done.